### PR TITLE
CR-19179-fix-task-order

### PR DESCRIPTION
**Pull request by ghost**
## What
improve ordering of handled tasks

## Why
in some edge cases, workflow termination tasks might attempt to run before the workflow creation was run, at the end result is that some pods will remain (in error state) in the cluster.

## Notes
this will resolve the issues that Monday had on June 7th, 2023

### CODE REVIEW

**Review by danielm-codefresh — APPROVED** [review]

File: venona/pkg/queue/queue.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		log         logger.Logger
+		wg          *sync.WaitGroup
+		monitor     monitoring.Monitor
+		queue       chan *workflow.Workflow
+		concurrency int
+		stop        []chan bool
+	}
+)
+
+const defaultWfTaskBufferSize = 20
```
Comment (danielm-codefresh) [inline venona/pkg/queue/queue.go:43]: Not sure where this is being used

File: venona/pkg/queue/queue.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+)
+
+// New creates a new TaskQueue instance
+func New(runtimes map[string]runtime.Runtime, log logger.Logger, wg *sync.WaitGroup, monitor monitoring.Monitor, concurrency int) *WorkflowQueue {
+	return &WorkflowQueue{
+		runtimes:    runtimes,
+		log:         log,
+		wg:          wg,
+		monitor:     monitor,
+		queue:       make(chan *workflow.Workflow, 1000),
```
Comment (danielm-codefresh) [inline venona/pkg/queue/queue.go:56]: Let's save it in a variable

File: venona/pkg/queue/queue.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+// Stop sends a signal to each of the handler to notify it to stop once the queue is empty
+func (wfq *WorkflowQueue) Stop() {
+	for i := 0; i < wfq.concurrency; i++ {
+		wfq.stop[i] <- true
+	}
+}
+
+func (wfq *WorkflowQueue) Size() int {
```
Comment (danielm-codefresh) [inline venona/pkg/queue/queue.go:80]: Can you please add a comment here as well?

File: venona/pkg/queue/queue.go (hunk 1 above)
Comment (noam-codefresh) [reply venona/pkg/queue/queue.go]: removed

File: venona/pkg/queue/queue.go (hunk 2 above)
Comment (noam-codefresh) [reply venona/pkg/queue/queue.go]: i exposed it as a cli flag (+ env variable `WORKFLOW_BUFFER_SIZE`), which defaults to 1000.
i hope that is what you meant.

**Pull request closed without merging**